src/plugins/data/common/query/to_expression_ast.ts

```diff
@@ -52,16 +54,21 @@ export async function queryStateToExpressionAst({
     filters: filters && filtersToAst(filters),
   });
   const ast = buildExpression([kibana, kibanaContext]).toAst();
-
   if (query && isOfAggregateQueryType(query)) {
     const mode = getAggregateQueryMode(query);
     // sql query
     if (mode === 'sql' && 'sql' in query) {
       const idxPattern = getIndexPatternFromSQLQuery(query.sql);
       const idsTitles = await dataViewsService.getIdsWithTitle();
-      const dataViewIdTitle = idsTitles.find(({ title }) => title === idxPattern);
-      if (dataViewIdTitle) {
-        const dataView = await dataViewsService.get(dataViewIdTitle.id);
+
+      let dataViewId = idsTitles.find(({ title }) => title === idxPattern)?.id;
+      // use ad-hoc data view if exists
+      if (!dataViewId && adHocDataView?.title === idxPattern) {
+        dataViewId = adHocDataView.id;
+      }
+
+      if (dataViewId) {
+        const dataView = await dataViewsService.get(dataViewId);
```

Comment:
Great that now also text based query langs work with AdHoc data views 🥳 , one note about the next level, out of scope of this PR. Use the query string to extract the index pattern and create the ad hoc one out of it, which would give the user a great flexibility, just enter whatever index pattern you like, no more prerequisites here (of course there should be matching indices, which needs to be checked). FYI @stratoula

Reply:
In general I think that would be nice but we need a way to specify the default timefield inline in this situation - otherwise the usefulness of the ad-hoc data view is limited.

Reply:
There have been ideas to put a field select into the time picker